### ISSUE TITLE
Failed to load NIF library

### ISSUE THREAD

**Issue by pobengtsson** [mdaiter/SpeakErlang#1]
Hi @mdaiter,

I was going to try your SpeakErlang integration and ran into a problem that haven't got around. The error message is:
{error,{{shutdown,{failed_to_start_child,speak,
                                         {bad_return_value,{{error,{load_failed,"Failed to load NIF library: 'dlopen(../priv/erl_espeak.so, 2): image not found'"}},
                                                            []}}}},
        {speak_app,start,[normal,[]]}}}

 Since I am still new to OTP apps, rebar3, etc it might be a silly mistake that is easy for you to spot and for me to fix.

reproduce like this:

in shell:
```
my_mac: ~ $ mkdir temp
my_mac: ~/temp $ rebar3 new app myapp
===> Writing myapp/src/myapp_app.erl
===> Writing myapp/src/myapp_sup.erl
===> Writing myapp/src/myapp.app.src
===> Writing myapp/rebar.config
===> Writing myapp/.gitignore
===> Writing myapp/LICENSE
===> Writing myapp/README.md
my_mac: ~/temp $ cd myapp/
```
I update the rebar.config....
```
POsMac:myapp po$ cat rebar.config 
{erl_opts, [debug_info]}.
{deps, [
{erl_speak, {git, "https://github.com/mdaiter/SpeakErlang.git"}}
]}.

my_mac: ~/temp/my_app $ rebar3 shell
===> Verifying dependencies...
===> Fetching erl_speak ({git,"https://github.com/mdaiter/SpeakErlang.git"})
===> WARNING: It is recommended to use {branch, Name}, {tag, Tag} or {ref, Ref}, otherwise updating the dep may not work as expected.
===> Compiling speak
_build/default/lib/erl_speak/src/speak.erl:2: Warning: undefined callback function handle_call/3 (behaviour 'gen_server')

===> Compiling myapp
Erlang/OTP 20 [erts-9.0] [source] [64-bit] [smp:8:8] [ds:8:8:10] [async-threads:0] [hipe] [kernel-poll:false] [dtrace]

Eshell V9.0  (abort with ^G)
1> application:start(speak).
{error,{{shutdown,{failed_to_start_child,speak,
                                         {bad_return_value,{{error,{load_failed,"Failed to load NIF library: 'dlopen(../priv/erl_espeak.so, 2): image not found'"}},
                                                            []}}}},
        {speak_app,start,[normal,[]]}}}
2> 
=INFO REPORT==== 11-Oct-2017::10:43:02 ===
    application: speak
    exited: {{shutdown,
                 {failed_to_start_child,speak,
                     {bad_return_value,
                         {{error,
                              {load_failed,
                                  "Failed to load NIF library: 'dlopen(../priv/erl_espeak.so, 2): image not found'"}},
                          []}}}},
             {speak_app,start,[normal,[]]}}
    type: temporary

2> 
```
I am on Mac Os X Sierra (not high sierra), but the same problem is reproducible on Raspberry Pi 3 with raspian/pixel, too.

Thanks for making SpeakErlang available and your time to read and respond to this issue!
/PO

**Comment by mdaiter** [comment]
Have you tried manually compiling those C files? Don't know if that .so gets generated by rebar in the current implementation.

It'd be a make command within the deps folder

**Comment by mdaiter** [comment]
Specifically, try going into the deps folder and search for espeak. In the espeak directory, go into c_src, and run make. That should give you the .so you're looking for

**Comment by pobengtsson** [comment]
thanks for your response. :)

The *.so files are compiled by rebar3 and placed in the priv directory. 
```
./_build/default/lib/erl_speak/priv:
total 56
-rwxr-xr-x  1 po  staff  10020 11 Okt 10:42 erl_espeak.so
-rwxr-xr-x  1 po  staff  15804 11 Okt 10:42 espeak_nif.so
```
So, to me, it looks like they are in the expected place, but some path setting I don’t really understand/know about is missing or pointing to the wrong place.

I have tried manually loading the nif with various paths in the rebar3 shell:
`erlang:load_nif(”path”,0).`
But get the same error. I also added a relx.config like below, but same error:
`{release, {my_app, "1"},
  [my_app, speak]}.`

Any other hint or thoughts are most appreciated!
